src/test/kotlin/no/nav/syfo/behandler/kafka/dialogmeldingfrombehandler/KafkaDialogmeldingFromBehandlerSpek.kt

```diff
@@ -26,41 +37,108 @@ class KafkaDialogmeldingFromBehandlerSpek : Spek({
-                        val dialogmelding = generateDialogmeldingFromBehandlerDTO(dialogmeldingMsgId)
-                        val dialogmeldingRecord = ConsumerRecord(
-                            DIALOGMELDING_FROM_BEHANDLER_TOPIC,
-                            partition,
-                            1,
-                            dialogmeldingMsgId.toString(),
-                            dialogmelding,
-                        )
-                        val mockConsumer = mockk<KafkaConsumer<String, KafkaDialogmeldingFromBehandlerDTO>>()
-                        every { mockConsumer.poll(any<Duration>()) } returns ConsumerRecords(
-                            mapOf(
-                                dialogmeldingTopicPartition to listOf(
-                                    dialogmeldingRecord,
-                                )
+                    it("update kontor with partnerID found in dialogmelding by setting dialogmeldingEnabled") {
+                        addBehandlerAndKontorToDatabase(behandlerService)
+
+                        val dialogmelding = generateDialogmeldingFromBehandlerDTO(UUID.randomUUID())
+                        val mockConsumer = mockKafkaConsumerWithDialogmelding(dialogmelding)
+
+                        runBlocking {
+                            pollAndProcessDialogmeldingFromBehandler(
+                                kafkaConsumerDialogmeldingFromBehandler = mockConsumer,
+                                database = database,
                             )
-                        )
-                        every { mockConsumer.commitSync() } returns Unit
+                        }
+                        verify(exactly = 1) { mockConsumer.commitSync() }
+
+                        val kontor = database.connection.getBehandlerKontorForPartnerId(UserConstants.PARTNERID)
+
+                        kontor?.dialogmeldingEnabled shouldNotBe null
+                    }
+                }
+
+                describe("Unhappy path") {
+                    it("don't update if partnerID in dialogmelding isn't found in kontor table") {
+                        val dialogmelding = generateDialogmeldingFromBehandlerDTO(UUID.randomUUID())
+                        val mockConsumer = mockKafkaConsumerWithDialogmelding(dialogmelding)
 
                         runBlocking {
                             pollAndProcessDialogmeldingFromBehandler(
                                 kafkaConsumerDialogmeldingFromBehandler = mockConsumer,
+                                database = database,
                             )
                         }
                         verify(exactly = 1) { mockConsumer.commitSync() }
+
+                        val kontor = database.connection.getBehandlerKontorForPartnerId(UserConstants.PARTNERID)
+
+                        kontor `should be` null
+                    }
+
+                    it("don't update kontor if we can't find partnerID in dialogmelding xml because it's invalid") {
```

Comment:
superpirk: _it's_  kan referere til flere objekter i setninga, så for å unngå tvetydighet kan man skrive seg rundt: Feks: "don't update kontor if we can't find partnerId because the dialogmelding xml is invalid".